### PR TITLE
JBPM-10199: Mark Async nodes as finished at the SVG when process inst…

### PR DESCRIPTION
**Pull request by nmirasch**
…ance is aborted

jira:https://issues.redhat.com/browse/JBPM-10199

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>a pull request</b> please add comment: <b>Jenkins retest this</b>
 
* <b>a full downstream build</b> please add comment: <b>Jenkins run fdb</b>
  
* <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

* <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

* <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>
</details>

<details>
<summary>
How to backport a pull request to a different branch?
</summary>

In order to automatically create a **backporting pull request** please add one or more labels having the following format `backport-<branch-name>`, where `<branch-name>` is the name of the branch where the pull request must be backported to (e.g., `backport-7.67.x` to backport the original PR to the `7.67.x` branch).

> **NOTE**: **backporting** is an action aiming to move a change (usually a commit) from a branch (usually the main one) to another one, which is generally referring to a still maintained release branch. Keeping it simple: it is about to move a specific change or a set of them from one branch to another.

Once the original pull request is successfully merged, the automated action will create one backporting pull request per each label (with the previous format) that has been added.

If something goes wrong, the author will be notified and at this point a manual backporting is needed.

> **NOTE**: this automated backporting is triggered whenever a pull request on `main` branch is labeled or closed, but both conditions must be satisfied to get the new PR created.
</details>

### CODE REVIEW

**Comment by nmirasch** [comment]
Jenkins run fdb

**Review by fjtirado — APPROVED** [review]

File: kie-server-parent/kie-server-services/kie-server-services-jbpm-ui/src/main/java/org/kie/server/services/jbpm/ui/ImageServiceBase.java [hunk 1]
```diff
@@ -144,8 +144,7 @@ public String getActiveProcessImage(String containerId, long procInstId, String
                     fullLogs.stream()
                             .filter(nodeInstanceDesc ->
                                             (((org.jbpm.kie.services.impl.model.NodeInstanceDesc) nodeInstanceDesc).getType() == NodeInstanceLog.TYPE_ASYNC_ENTER) &&
-                                                    fullLogs.stream().noneMatch(nodeInst -> nodeInstanceDesc.getNodeId().equals(nodeInst.getNodeId())
-                                                            && (((org.jbpm.kie.services.impl.model.NodeInstanceDesc) nodeInst).getType() == NodeInstanceLog.TYPE_EXIT)))
```
Comment (domhanak) [inline kie-server-parent/kie-server-services/kie-server-services-jbpm-ui/src/main/java/org/kie/server/services/jbpm/ui/ImageServiceBase.java:148]: @nmirasch @fjtirado could you please share a brief explanation about why we did check by type also?

Comment (domhanak) [reply kie-server-parent/kie-server-services/kie-server-services-jbpm-ui/src/main/java/org/kie/server/services/jbpm/ui/ImageServiceBase.java:148]: I see this comment ` // Async active nodes don't have any related completed node instance`

Comment (nmirasch) [reply kie-server-parent/kie-server-services/kie-server-services-jbpm-ui/src/main/java/org/kie/server/services/jbpm/ui/ImageServiceBase.java:148]: @domhanak before this change it only checked that the async node has an type_exit log to include it as an active one. Now it checks that it's not in the finished ones (including Exited, Aborted, Skipped and Obsolete) to be included as an active node and painted properly.

**Comment by domhanak** [comment]
jenkins do fdb

**Comment by domhanak** [comment]
@nmirasch My current end result is this:
image

Could you please confirm this is expected? I am a  bit confused as the issue states the Human Task should have Blue Borders but it doesn't

**Comment by nmirasch** [comment]
@domhanak the weird result that I wasn't able to reproduce, is that blue border color (active nodes) after aborting the instance. I tried aborting it via rest and from the BC with the same results, no blue color anywhere. The expected node colors at the diagram are:
- Border Red: active Async nodes
- Border Blue: active nodes
- Gray filling and dark border: completed. 
If you have more details about how to reproduce the issue you found would be great, Thanks in advanced!

**Comment by domhanak** [comment]
@nmirasch Okay, I will give it another go and let you know.

**Review by domhanak — APPROVED** [review]
I have confirmed with @nmirasch that this  `blue bordrer` on End state is not related and seems to be a separate issue. After fresh install, I am unable to reproduce it again.

Apologies for the weird issues report but I was able to do reproduce it consistently and did so several times before posting here :D

Approved.

**Comment by nmirasch** [comment]
thanks @domhanak !

**Pull request merged**